### ISSUE TITLE
Noisy finger joint encoder readings

### ISSUE THREAD

**Issue by christian-rauch** [openhumanoids/wiki#1]
We observed that the joint encoder readings for at least the fingers on the left hand are noisy. E.g. the fingers visualised in director are moving with a high frequency when they are not actuated.

The frequency spectrum of the reported joint positions (CORE_ROBOT_STATE) shows a high periodicity at `12.3 Hz` and `12.3 +/- 1 Hz` for most joints.

Spectrum for all finger joints: [finger_joint_frequency_spectrum.zip](https://github.com/openhumanoids/wiki/files/462005/finger_joint_frequency_spectrum.zip).

Exemplary for first joint on middle finger and pinky:

| joint | spectrum (without 0Hz) |
| --- | --- |
| leftMiddleFingerPitch1 | ![leftmiddlefingerpitch1](https://cloud.githubusercontent.com/assets/8226248/18355683/ccb035fe-75e1-11e6-9673-aa3766a2ccb2.png) |
| leftPinkyPitch1 | ![leftpinkypitch1](https://cloud.githubusercontent.com/assets/8226248/18355725/f60d68c2-75e1-11e6-83f4-e0f705df3618.png) |

We are currently using an alpha filter (https://github.com/openhumanoids/oh-distro-private/issues/2452) to reduce the noise.

Is this issue known? Is this a hardware issue (e.g., power cables next to data cables) that also affects other joint encoders?

**Comment by jjholley** [comment]
Yes, these sensors are known to be noisy.  Can you provide details on how this data was collected?  I would like to reproduce this data on fingers here.

**Comment by christian-rauch** [comment]
The left hand was moved and then stayed for 175 seconds in this pose:
![val_pose_finger_joint_collection](https://cloud.githubusercontent.com/assets/8226248/18358857/28770f72-75ef-11e6-980b-28a1cc507626.png)

For generating the spectrum, I used numpy's fft  module. I specifically used the script `plot_spectrum.py` from https://github.com/christian-rauch/lcm_joint_state_spectrum with the path to the logfile as first argument.

**Comment by jjholley** [comment]
Was this done on the robot?  On a testbench?  What logger did you use to collect the data?

**Comment by christian-rauch** [comment]
We were running the robot with attached hand. The analysed values are from the LCM channel CORE_ROBOT_STATE which was logged through the logger that is integrated in director.

**Comment by mauricefallon** [comment]
Basically its a ROS controller thats reading the state of the fingers and publishes messages to our messaging system, LCM at 500Hz (I think).

We can give you the data as a csv if you wish

@christian-rauch : folks in NASA dont know the internal details of our system!

**Comment by jjholley** [comment]
If you don't mind sending me a csv, I would appreciate it.

**Comment by christian-rauch** [comment]
I uploaded the csv to our server: http://terminator.robots.inf.ed.ac.uk/bignas/useful/lcm/20160816-cr_grasping/noisy_finger_joints.csv. The first column gives the time in seconds, the remaining columns are joint positions in radiant.
